### PR TITLE
Autorotate xAxisLabel to 45˚ 

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

**Comment by simzer** [comment]
Tested it a bit, found problems:
- auto rotation value is a more than 45 deg (possibly because angle param is in radian, but set as deg. 45rad = 58,31 + n*180 deg)
- auto rotation overrides the explicitly set angle style parameter, which means that the user cannot override the auto value
- it labels are auto-rotated but the next target chart does not need label rotation, it still will have rotated labels, and rotation will be switched off only after the next anim.

**Comment by schaumb** [comment]
Fixed:
> auto rotation value is a more than 45 deg (possibly because angle param is in radian, but set as deg. 45rad = 58,31 + n*180 deg)

---

> should not rotate the x axis labels in polar coordinate system

It'll not rotate on polar coordinate system, because of [else if](https://github.com/vizzuhq/vizzu-lib/pull/454/files#diff-9759f752575ebdb37254436cf9fe4d6e31a2d5632094d0b998d8931b496f86ccR92).

**Comment by schaumb** [comment]
The xAxis label default left/right padding will be 6/12, based on the latest commit tests. I will go through all failing test cases, and I will create a new test for this feature.

**Comment by schaumb** [comment]
Conclusion: The current padding is not a good measure for checking the texts to be close to each other, because currently, it contains the axis "padding" too. Because of that (and the axis padding will be hard to implement), we thought that the easiest is to measure half of the padding.

**Comment by simzer** [comment]
> * it labels are auto-rotated but the next target chart does not need label rotation, it still will have rotated labels, and rotation will be switched off only after the next anim.

It works now:
https://jsfiddle.net/simzer/am7L0uwo/7/

**Review by simzer — APPROVED** [review]

File: test/e2e/tests/config_tests/x_axis_label_autorotate.mjs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const series = [
```
[the remark this answers is not in the document]
Comment (schaumb) [reply test/e2e/tests/config_tests/x_axis_label_autorotate.mjs:1]: I'll create a new PR for this

[the remark this answers is not in the document]
Comment (schaumb) [reply test/e2e/tests/config_tests/x_axis_label_autorotate.mjs:1]: https://github.com/vizzuhq/vizzu-lib/pull/471

File: src/apps/weblib/interface.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
 mergeInto(LibraryManager.library, {
 	openUrl: function (url) {
 		window.open(UTF8ToString(url), '_blank')
+	},
+	textBoundary: function (font, text, sizeX, sizeY) {
```
[the remark this answers is not in the document]
Comment (schaumb) [reply src/apps/weblib/interface.js:5]: Not resolved. We need to fix it [here](https://github.com/vizzuhq/vizzu-lib/issues/165)

File: src/apps/weblib/ts-api/module/module.ts [hunk 3]
```diff
@@ -12,6 +12,9 @@ import { Canvas } from './canvas'
 export class Module extends CEnv {
 	constructor(wasm: CVizzu) {
 		super(wasm, new ObjectRegistry(wasm._object_free))
+		const context2D = (<HTMLCanvasElement>document.createElement('canvas')).getContext('2d')
```
[the remark this answers is not in the document]
Comment (schaumb) [reply src/apps/weblib/ts-api/module/module.ts:15]: Not resolved. We need to fix it [here](https://github.com/vizzuhq/vizzu-lib/issues/165)

**Pull request merged**